multiaddr/protocols.py

```diff
@@ -151,7 +151,7 @@ def read_varint_code(buf):
     Protocol(P_UDP, 'udp', 'uint16be'),
     Protocol(P_DCCP, 'dccp', 'uint16be'),
     Protocol(P_IP6, 'ip6', 'ip6'),
-    Protocol(P_IP6ZONE, 'ip6zone', '?'),
+    Protocol(P_IP6ZONE, 'ip6zone', 'ip6zone'),
```

Comment:
The approach LGTM!

I am not sure however this codec should be named “ip6zone” if all it does is decoding stuff from/to UTF-8. I know `p2p` is basically the same with base58btc, but still…
The question I'm basically asking is: Is this usage of UTF-8-encoded strings special enough to be considered ip6zone-specific?
Thoughts on this @mhchia?

Reply:
Good point. The encoding shouldn't be limited to ip6zone. From [go implementation](https://github.com/multiformats/go-multiaddr/blob/0460aad0fe51f7aa64d3a3c17854dc25368f7ca2/transcoders.go#L62-L85), I can only tell the coding for ip6zone is "encoded/decoded with UTF-8" by golang and "the content should not be zero-lengthed". Will it be better to be renamed to something like "not-empty-utf-8"? or do you have suggestions on this?

Reply:
Maybe just “utf8”? Only fspath currently allows an empty value and that wasn't a conscious decision on my end and empty paths are invalid (at least on Linux) so it's not like it would make sense anyway. If we'd ever need to distinguish between these cases we could still relax this one and add a more specific one.

Reply:
Reasonable. Thank you for the suggestion. I will change the name. :+1: